### PR TITLE
Random sample data

### LINKED ISSUE #27: Create constructor for random time series
User supplies the date range and the method uses a `RNG` to create a random walk time series.

### PR DESCRIPTION
**Pull request by femtotrader**
Closes #27

### CODE REVIEW

File: src/random.jl [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    start = Dates.DateTime(2020, 1, 1),
+    step = Dates.Minute(1),
+    length = 500,
+    price_init = nothing,
+    price_init_min = 1.00,
+    price_init_step = 0.01,
+    price_init_max = 1000.00,
+    price_var_min = -1.0,
+    price_var_step = 0.01,
+    price_var_max = 0.01,
```
Comment (Arkoniak) [inline src/random.jl:11]: Is downward trend intentional? Maybe by default should be `price_var_max = 1.0`?

File: src/random.jl [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        step = sub_step,
+        length = length * nsub,
+        price_init = price_init,
+        price_init_min = price_init_min,
+        price_init_step = price_init_step,
+        price_init_max = price_init_max,
+        price_var_min = price_var_step,
+        price_var_max = price_var_max,
+    )
+    price
```
Comment (Arkoniak) [inline src/random.jl:69]: This one probably is a leftover?

**Comment by Arkoniak** [comment]
It would be nice to have tests, but for this it's better to add `rng` parameter in function signature.

For example:
```
using Random

function random_cl(rng::AbstractRNG = Random.default_rng();
    start = Dates.DateTime(2020, 1, 1),
    ... # all other fields
```

and use it internally with constructions like
```
price_init .+ cumsum(rand(rng, price_var_min:price_var_step:price_var_max, length))
```

I am not sure though, if it is possible to use `Random.default_rng()` in Julia versions prior to 1.9

File: src/random.jl (hunk 1 above)
Comment (femtotrader) [reply src/random.jl]: Fixed

File: src/random.jl (hunk 2 above)
Comment (femtotrader) [reply src/random.jl]: Fixed also

**Comment by femtotrader** [comment]
Tests have been added
Should be ready now

**Pull request merged**